collection-scripts/gather

```diff
@@ -16,6 +16,12 @@ for localns in $(/usr/bin/oc get migrationcontrollers.migration.openshift.io --a
   done
   echo "Will collect debug info from migclusters [${clusters[@]}]"
 
+  # Find the latest migration, plan, and associated namespaces so we can gather additional info from those
+  latest_migration=$(oc -n ${localns} get migmigration -o json | jq -r '.items|=sort_by(.metadata.creationTimestamp)' | jq -r '.items[-1].metadata.name')
```

Comment:
I am not familliar with jq. Does items[-1] fail when there are no migmigrations present?

Reply:
That fetches the last entry in items, which will be the item created in the last based on the sort I think. Correct me if I am misinterpreting this @djwhatle

Reply:
Looks like it results in "null" getting set to the var and a 0 return code. Will try a full must-gather in this scenario and make sure it doesn't crash.

```
$ oc delete migmigration --all
migmigration.migration.openshift.io "8069a3d0-8c19-11eb-b0bf-5fcc05ad4e8c" deleted
migmigration.migration.openshift.io "b5a018a0-8b1e-11eb-ac4b-3b5c0d147d47" deleted

latest_migration=$(oc -n openshift-migration get migmigration -o json | jq -r '.items|=sort_by(.metadata.creationTimestamp)' | jq -r '.items[-1].metadata.name')

$  echo $?
0
$ echo $latest_migration
null
```

Reply:
Ok, so basically the end-result is that all of the `latest_*` vars get set to `null`, and inside the loop that compares current cluster name to `latest_cluster`, we'll never enter the new block for gathering content of migrated namespaces.

Output from must-gather
```
[must-gather-n94dc] POD Error from server (NotFound): migmigrations.migration.openshift.io "null" not found
[must-gather-n94dc] POD jq: error (at <stdin>:191): Cannot iterate over null (null)
[must-gather-n94dc] POD [cluster=host][namespace=openshift-migration] Detected MTC installation
[must-gather-n94dc] POD [cluster=host] Not found within latest_migration_clusters=null
[must-gather-n94dc] POD null
```

So I think this is harmless and works as-is, however I'm open to input on whether we should squash the jq stdout and send it to /dev/null so that the error never shows up.

Reply:
@djwhatle I was under impression that if a script in must-gather returns non-zero exit code, the intermediary pod exits with non-zero status. But maybe that is incorrect. I think it'd be OK to have stderr printed in this way since this doesn't really break the must-gather container.

Reply:
You're correct, it's just that the RC is always 0 even when `jq` doesn't find something.